### ISSUE TITLE
OpenShot Animated Title Rendering

### ISSUE THREAD

**Issue by bbacle** [OpenShot/openshot-qt#3268]
**Describe the bug**
Animated Titles rendering

**Steps to reproduce the behavior:**
1. Go to "Title - Animated Title"
2. Click on "Dissolving Text"
3. Input "Gilligan" for the "File Name"
4. Input "Gilligan's Island" in "Title"
5. Input "1.50" for the "Text Size"
6. Try changing  Fonts in the "Font Name" (Nothing works there except the Bfont) - Why is this?
7. Slide the slider back to frame 1
8. Click on Render
9. The rendering only goes to frame 13 and stops (I left it on all night and it was still on frame 13 the next day)
10. I have gone back to version 2.4 and it rendered the animated title.

**Expected behavior**
I expected it to render the title for me so that I could put it into the timeline and export it to use in another video editing program that I use all the time.
It did not render in version 2.5 and in version 2.5.1 - it only worked in version 2.4.
Also I could not change to any other font (which, by the way, is very limited) - Can you not make it where it will accept any font that I have installed on my computer?
So there is the font issue as well as the rendering issue.
BTW...the font issue is on all versions 2.4 and forward.

**System Details**
- OpenShot Version: 2.5.1
- libopenshot Version:  0.2.5
- Platform: Windows-7-6.1.7601-SP1
- Processor: Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
- Machine: AMD64
- Python version: 3.7.4
- Qt5 version: 5.12.4
- PyQt5 version: 5.12.3
- Qt Detected Languages: ['en-US']
- LANG Environment Variable: 
- LOCALE Environment Variable: 
- Daily Build Installed: http://github.com/OpenShot/openshot-qt/releases/download/OpenShot-v2.5.1-release-candidate-1583026317-x86_64.exe

**Log Files**
- http://logs.openshot.org/730u3oip38wca5kvefjbez8pt74d319t/openshot-qt.zip
- http://logs.openshot.org/730u3oip38wca5kvefjbez8pt74d319t/libopenshot.zip

**Exception / Stacktrace**
```No stacktrace found in log files```

**Screenshots** (Optional)
If applicable, add screenshots to help explain your problem. You can include screenshots by
copy/pasting them on GitHub or dragging-and-dropping into the GitHub page. All images are public,
so please don't post screenshots containing personal information.

**Comment by ferdnyc** [comment]
Hmm. Looking at your logs, the Blender job appears to have gone just as you said: It rendered fine for the first 13 frames, and then simply stopped... the next event is you closing the window.
```
blender_listview:INFO Using parameter file_name: File Name
blender_listview:INFO Using parameter title: Title
blender_listview:INFO Using parameter extrude: Extrude
blender_listview:INFO Using parameter bevel_depth: Bevel Depth
blender_listview:INFO Using parameter fontname: Font Name
blender_listview:INFO Animation param fontname set to Bfont
blender_listview:INFO Using parameter spacemode: Text Alignment
blender_listview:INFO Animation param spacemode set to CENTER
blender_listview:INFO Using parameter text_size: Text Size
blender_listview:INFO Using parameter width: Text Width
blender_listview:INFO Using parameter diffuse_color: Diffuse Color
blender_listview:INFO Using parameter start_frame: Start Frame
blender_listview:INFO Using parameter end_frame: End Frame
blender_listview:INFO Using parameter animation_speed: Animation Length
blender_listview:INFO Animation param animation_speed set to 2
blender_listview:INFO Animation param animation_speed set to 1
blender_listview:INFO Previewing frame 64
blender_listview:INFO Blender command: C:/Program Files/Blender Foundation/Blender 2.82/blender.exe -b 'C:\Program Files\OpenShot Video Editor\blender\blend\dissolve.blend' -P 'C:\Users\Airwolf\.openshot_qt\blender\3VDA8MFO86\dissolve.py'
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\TitleFileName0064.png
blender_listview:INFO Previewing frame 1
blender_listview:INFO Blender command: C:/Program Files/Blender Foundation/Blender 2.82/blender.exe -b 'C:\Program Files\OpenShot Video Editor\blender\blend\dissolve.blend' -P 'C:\Users\Airwolf\.openshot_qt\blender\3VDA8MFO86\dissolve.py'
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\TitleFileName0001.png
blender_listview:INFO Animation param text_size set to 1.0
blender_listview:INFO Animation param text_size set to 1.5
blender_listview:INFO Previewing frame 1
blender_listview:INFO Blender command: C:/Program Files/Blender Foundation/Blender 2.82/blender.exe -b 'C:\Program Files\OpenShot Video Editor\blender\blend\dissolve.blend' -P 'C:\Users\Airwolf\.openshot_qt\blender\3VDA8MFO86\dissolve.py'
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0001.png
blender_listview:INFO Blender command: C:/Program Files/Blender Foundation/Blender 2.82/blender.exe -b 'C:\Program Files\OpenShot Video Editor\blender\blend\dissolve.blend' -P 'C:\Users\Airwolf\.openshot_qt\blender\3VDA8MFO86\dissolve.py'
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0001.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0002.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0003.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0004.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0005.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0006.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0007.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0008.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0009.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0010.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0011.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0012.png
blender_listview:INFO Image detected from blender regex: C:\\Users\\Airwolf\\.openshot_qt\\blender\\3VDA8MFO86\\Gilligan0013.png
 main_window:INFO animated title add cancelled
 main_window:INFO ---------------- Shutting down -----------------
```

...It's possible there's a compatibility issue with Blender 2.82, the code was only tested with 2.80. (But if you say OpenShot 2.4.4 worked, I assume with 2.80, then that's very strange. AFAIK the animated-titles code is completely identical between the two versions, though I'll double-check that.)

It's _possible_ it was just a random Blender hang, and re-attempting the render in 2.5.1 would've eventually succeeded as well.

**Comment by ferdnyc** [comment]
Oh, no, sorry — my mistake. (Old age setting in.) I misremembered the release history.

v2.4.4 uses Blender <= 2.79, which has a completely different API from Blender 2.80+, which is compatible with OpenShot starting from 2.5.0. So, the _OpenShot_ code is pretty much the same, but the _Blender_ animation code is completely different between 2.4.4 and 2.5.1.

It's possible there's an issue with the updated version of `dissolve.blend`.

I'll install Blender 2.82 and test it out. If you'd like to take a look yourself, when you start the render a line is logged to the `%USERPROFILE%/.openshot_qt/openshot-qt.log` file like the following:
```
C:/Program Files/Blender Foundation/Blender 2.82/blender.exe 
-b 'C:\Program Files\OpenShot Video Editor\blender\blend\dissolve.blend' 
-P 'C:\Users\Airwolf\.openshot_qt\blender\90D5J4LFCH\dissolve.py'
```
(The pathnames will change, since they're autogenerated.) Running that same command in a Windows CMD prompt, and adding these command line options to the end, might give useful output:
```
--verbose 2 --debug-python --debug-memory --debug-jobs
```

**Comment by ferdnyc** [comment]
Actually, I have a theory on what might be going on here.

The `.blend` file _appears_ to be generating a bunch of error messages for each frame rendered — those messages seem to be harmless, **but** because the process' `stderr` output is piped to to the calling thread, and then never read, eventually those messages will fill up the buffer and [cause a deadlock](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.stderr).

The **right** solution to this is to use `Popen.communicate()` to poll the process, but for now I'm going to see if we can get away with just redirecting to the output stream with `stderr=subprocess.STDOUT`. (That'll also let us potentially capture and/or display the messages, if need be.)

**Comment by ferdnyc** [comment]
@bbacle 

This OpenShot installer:

https://drive.google.com/file/d/1MP804iCSGGHX-GmuBxHHfUv8JjlSA0_v/view?usp=sharing

is built from the branch for my PR #3273, which (I hope) fixes the way Blender is executed so that it will no longer hang while rendering. Would you be able to try it out and let me know if it fixes the issue? Thanks.

**Comment by jonoomph** [comment]
Fix will be available on our daily builds very soon: https://www.openshot.org/download/#daily

**Comment by ferdnyc** [comment]
(Yeah, don't download my installer above. IT BROKEN.)

**Comment by Derek-K** [comment]
Confirm _OpenShot-v2.5.1-x86_64_ stable build, "Dissolving Text" does stuck in frame 13.
Tested with _OpenShot-v2.5.1-dev2-1590649087-10ed4baa-7cec1822-x86_64_ and the original issue is now fixed.
As for #3269 , also confirm only Bfont works correctly.

Just my .02

**Comment by KDesigned** [comment]
Confirm OpenShot-v2.5.1-x86_64 stable build, "Dissolving Text" does stuck in frame 13.
I confirm this issue, v2.5.1 dev2 installed, Blender 2.8.3 installed, NONE of the animated texts works, Windows shows the circling icon means it keeps trying to load.

**Comment by ferdnyc** [comment]
> Confirm OpenShot-v2.5.1-x86_64 stable build, "Dissolving Text" does stuck in frame 13.
> I confirm this issue, v2.5.1 dev2 installed, Blender 2.8.3 installed, NONE of the animated texts works, Windows shows the circling icon means it keeps trying to load.

@KDesigned 

That issue should be fixed in the daily builds, I'm not aware of any reasons you'd still be seeing it. You'll probably need to supply logs for us to investigate.

It could be that 2.83 introduces some new issue, I don't think we've tested with that version yet.

**Comment by KDesigned** [comment]
> > Confirm OpenShot-v2.5.1-x86_64 stable build, "Dissolving Text" does stuck in frame 13.
> > I confirm this issue, v2.5.1 dev2 installed, Blender 2.8.3 installed, NONE of the animated texts works, Windows shows the circling icon means it keeps trying to load.
> 
> @KDesigned
> 
> That issue should be fixed in the daily builds, I'm not aware of any reasons you'd still be seeing it. You'll probably need to supply logs for us to investigate.
> 
> It could be that 2.83 introduces some new issue, I don't think we've tested with that version yet.

**Describe the bug**
A clear and concise description of the issue

**Steps to reproduce the behavior:**
1. Go to 'Title'
2. Click on 'Animated Title'
3. Scroll down to 'any available preset'
4. does not stop displaying the circling load icon, selected (any) preset does not load in preview, cannot render

**Expected behavior**
Load the selected title and render. 

**System Details**
- OpenShot Version: 2.5.1-dev2
- libopenshot Version:  0.2.5-dev2
- Platform: Windows-10-10.0.18362
- Processor: Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
- Machine: AMD64
- Python version: 3.7.4
- Qt5 version: 5.12.4
- PyQt5 version: 5.12.3
- Qt Detected Languages: ['en-US']
- LANG Environment Variable: 
- LOCALE Environment Variable: 
- Daily Build: Verified issue still exists in daily build: http://github.com/OpenShot/openshot-qt/releases/download/OpenShot-v2.5.1-dev2-1592475299-517f2896-7cec1822-x86_64.exe

**Log Files**
- [openshot-qt.log](https://s3.amazonaws.com/logs.openshot.org/zp67lzsqa51f5yg2gqqkj1vm4b45bpf9/openshot-qt.zip) (15 KB)
- [libopenshot.log](https://s3.amazonaws.com/logs.openshot.org/zp67lzsqa51f5yg2gqqkj1vm4b45bpf9/libopenshot.zip) (6 KB)

**Exception / Stacktrace**
```No stacktrace found in log files```

**Comment by lalowhitney** [comment]
Hi There, I noticed this was marked as closed and this is basically the only thread I can find on the issue. I'm running into the exact issue described above by @KDesigned . 

I'm new to using Openshot and had installed the latest version from the site 2.5.1, then installed the latest version of blender from blender.org and was getting stuck at the 13th frame during render.

After reading this thread I had tried the latest daily build OpenShot-v2.5.1-dev2-1592475299-517f2896-7cec1822-x86_64, and now I get the spinning wheel of doom just making a title style selection where previously this would happen at the 13th frame of rendering.

Attached are the files.
[openshot-qt.log](https://github.com/OpenShot/openshot-qt/files/4847850/openshot-qt.log)
[libopenshot.log](https://github.com/OpenShot/openshot-qt/files/4847851/libopenshot.log)

Any help is appreciated.

**Comment by bbacle** [comment]
Hello...I was the one who originally posted this problem and it has still not been fixed, even though I have installed the latest version of OpenShot & Blender. I had to go back & install OpenShort 2.4.1 & Blender 2.79 to get it to work. Only the Bfont works too. I have asked them to make all fonts on my computer available, but still no dice. The Bfont is the only one that works.
I don't know if they are really listening, as they don't seem to do anything about this problem.

**Comment by lalowhitney** [comment]
> Hello...I was the one who originally posted this problem and it has still not been fixed, even though I have installed the latest version of OpenShot & Blender. I had to go back & install OpenShort 2.4.1 & Blender 2.79 to get it to work. Only the Bfont works too. I have asked them to make all fonts on my computer available, but still no dice. The Bfont is the only one that works.
> I don't know if they are really listening, as they don't seem to do anything about this problem.

I was able to get this to work by installing this daily build "OpenShot-v2.5.1-dev2-1592475299-517f2896-7cec1822-x86_64" and installing Blender 2.80. There's a lot of bad information out there. For example, if you visit their Reddit page, under the right rail is a "common problems and solution" section that says you need to set the blender path within Openshot. The example they give is just to a folder (C:\Program Files\Blender Foundation\Blender\). It needs to point to an executable. (C:\Program Files\Blender Foundation\Blender\blender.exe)

Either way, after a lot uninstalling and reinstalling and troubleshooting, I was able to get it work. But yeah only Bfont works!

Bottom line, if someone else comes across this post in hopes of finding a fix for Windows, the answer is to install their "daily Build" here:
https://github.com/OpenShot/openshot-qt/releases/download/daily/OpenShot-v2.5.1-dev2-1593459788-517f2896-7cec1822-x86_64.exe

and install Blender 2.80 from here:
https://www.blender.org/download/releases/2-80/

**Comment by bbacle** [comment]
That's great that you got it to work with a later version. I have not installed the later versions because I just kept getting the same problem as before. I finally just used an older version that worked for me. I will try the versions you recommend and see if it works. Thank you for the info.

**Comment by KDesigned** [comment]
**lalowhitney** thanks for the details, still does not work for me.
here a zip file containing a short video showing settings and the results

[Not Working-1.zip](https://github.com/OpenShot/openshot-qt/files/4849870/Not.Working-1.zip)

**Comment by KiLLAAA** [comment]
**Hello everyone, i have sorted it out..**
_it is caused by incorrect "loading" of fonts in python scripts related to blend files_

**steps to reproduce:**
1/ mount appimage
`mkdir /tmp/mountpoint`
`sudo mount -o loop OpenShot-v2.5.1-dev2-1595950874-bf0159bb-7cec1822-x86_64.AppImage /tmp/mountpoint`

2/ copy the image's content to new location where you have permissions to edit files, like /home/. . .

3/ navigate to your unpacked appimage copy's folder `/usr/bin/blender/scripts`

4/ edit the file `wireframe_text.py` (do not touch rest of the file)

`# Get font object
font = None
if params["fontname"] != "Bfont":
    # Add font so it's available to Blender
    #font = load_font(params["fontname"]) # <------------------------- comment out
    font = bpy.data.fonts.load("/home/k/.fonts/Ubuntu.ttf") # <------ add this
else:
    # Get default font
    font = bpy.data.fonts["Bfont"]`

5/ navigate to top directory of your unpacked copy and run it as root to avoid another bug with permissions
`sudo ./AppRun`

6/ click Title, Animated Title and in the newly opened window scroll down to Wireframe Text and click it

7/ pick any font except for the Bfont and move "Frame" slider to right

**RESULT: Preview rendered correctly with Ubuntu font!**

**Comment by KDesigned** [comment]
That sounds brilliant. Now I just need to have someone to tell me what to change in the windows version.
I found the said file in C:/Program Files/OpenShot Video Editor/blender/scripts

**Comment by KiLLAAA** [comment]
Well, i haven't found how to fix it (yet), just sorted out what causes this behavior..
The script which is being injected parameters from main UI and after that parsed by blender has problematic function "load_font(font_path)" which always returns "None", because only font in list provided by Blender bpy is Bfont. So the render font is None when anything other than Bfont gets selected...
Only way i found it working is to load font with full path specified, like `font = bpy.data.fonts.load("/path/to/font.ttf")`

**Comment by KiLLAAA** [comment]
@jonoomph **this is a bug, and has not been fixed, i suppose that it should be reopened...**

**Comment by KDesigned** [comment]
@ferdnyc please have a look and re-open this thread to allow for investigation / rectification, thanks

**Comment by KiLLAAA** [comment]
i compared version i use as a "production" **1.4.3** and found out that it provides full path, in recent versions i tried(2.5.1), even the daily build, _only font name is being injected to python script_. This may work on windows, but on linux **full path is required** for fonts..

_1.4.3 + Blender 2.69_
- full list of system wide and user fonts to select from
- full path provided for selected font name
`#BEGIN INJECTING PARAMS`
...
`params['fontname'] = '/home/k/.fonts/Ubuntu.ttf'`
...
`#END INJECTING PARAMS`

_2.5.1-dev2 + Blender 2.80_
- just a few fonts to pick from - system wide at location `/usr/share/fonts/truetype/freefont`
- only name of selected font provided = the problem why it doesn't work
`#BEGIN INJECTING PARAMS`
...
`params['fontname'] = u'FreeSans'`
...
`#END INJECTING PARAMS`

**Comment by helfgott** [comment]
> That sounds brilliant. Now I just need to have someone to tell me what to change in the windows version. I found the said file in C:/Program Files/OpenShot Video Editor/blender/scripts

Maybe too late but useful for someone else, I made a workaround here for windows: https://github.com/OpenShot/openshot-qt/issues/3269#issuecomment-1636618140
